test/parallel/test-http2-binding.js

```diff
@@ -0,0 +1,229 @@
+  HTTP2_HEADER_ACCEPT_LANGUAGE: 'accept-language',
+  HTTP2_HEADER_ACCEPT_RANGES: 'accept-ranges',
+  HTTP2_HEADER_ACCEPT: 'accept',
+  HTTP2_HEADER_ACCESS_CONTROL_ALLOW_ORIGIN: 'access-control-allow-origin',
+  HTTP2_HEADER_AGE: 'age',
+  HTTP2_HEADER_ALLOW: 'allow',
+  HTTP2_HEADER_AUTHORIZATION: 'authorization',
+  HTTP2_HEADER_CACHE_CONTROL: 'cache-control',
+  HTTP2_HEADER_CONTENT_DISPOSITION: 'content-disposition',
+  HTTP2_HEADER_CONTENT_ENCODING: 'content-encoding',
+  HTTP2_HEADER_CONTENT_LANGUAGE: 'content-language',
+  HTTP2_HEADER_CONTENT_LENGTH: 'content-length',
+  HTTP2_HEADER_CONTENT_LOCATION: 'content-location',
+  HTTP2_HEADER_CONTENT_RANGE: 'content-range',
+  HTTP2_HEADER_CONTENT_TYPE: 'content-type',
+  HTTP2_HEADER_COOKIE: 'cookie',
+  HTTP2_HEADER_CONNECTION: 'connection',
+  HTTP2_HEADER_ETAG: 'etag',
+  HTTP2_HEADER_EXPECT: 'expect',
+  HTTP2_HEADER_EXPIRES: 'expires',
+  HTTP2_HEADER_FROM: 'from',
+  HTTP2_HEADER_HOST: 'host',
+  HTTP2_HEADER_IF_MATCH: 'if-match',
+  HTTP2_HEADER_IF_MODIFIED_SINCE: 'if-modified-since',
+  HTTP2_HEADER_IF_NONE_MATCH: 'if-none-match',
+  HTTP2_HEADER_IF_RANGE: 'if-range',
+  HTTP2_HEADER_IF_UNMODIFIED_SINCE: 'if-unmodified-since',
+  HTTP2_HEADER_LAST_MODIFIED: 'last-modified',
+  HTTP2_HEADER_LINK: 'link',
+  HTTP2_HEADER_LOCATION: 'location',
+  HTTP2_HEADER_MAX_FORWARDS: 'max-forwards',
+  HTTP2_HEADER_PREFER: 'prefer',
+  HTTP2_HEADER_PROXY_AUTHENTICATE: 'proxy-authenticate',
+  HTTP2_HEADER_PROXY_AUTHORIZATION: 'proxy-authorization',
+  HTTP2_HEADER_PROXY_CONNECTION: 'proxy-connection',
+  HTTP2_HEADER_RANGE: 'range',
+  HTTP2_HEADER_REFERER: 'referer',
+  HTTP2_HEADER_REFRESH: 'refresh',
+  HTTP2_HEADER_RETRY_AFTER: 'retry-after',
+  HTTP2_HEADER_SERVER: 'server',
+  HTTP2_HEADER_SET_COOKIE: 'set-cookie',
+  HTTP2_HEADER_STRICT_TRANSPORT_SECURITY: 'strict-transport-security',
+  HTTP2_HEADER_TRANSFER_ENCODING: 'transfer-encoding',
+  HTTP2_HEADER_USER_AGENT: 'user-agent',
+  HTTP2_HEADER_VARY: 'vary',
+  HTTP2_HEADER_VIA: 'via',
+  HTTP2_HEADER_WWW_AUTHENTICATE: 'www-authenticate',
+  HTTP2_HEADER_KEEP_ALIVE: 'keep-alive',
+  HTTP2_HEADER_CONTENT_MD5: 'content-md5',
+  HTTP2_HEADER_TE: 'te',
+  HTTP2_HEADER_UPGRADE: 'upgrade',
+  HTTP2_HEADER_HTTP2_SETTINGS: 'http2-settings'
+};
+
+const expectedNGConstants = {
+  NGHTTP2_SESSION_SERVER: 0,
+  NGHTTP2_SESSION_CLIENT: 1,
+  NGHTTP2_STREAM_STATE_IDLE: 1,
+  NGHTTP2_STREAM_STATE_OPEN: 2,
+  NGHTTP2_STREAM_STATE_RESERVED_LOCAL: 3,
+  NGHTTP2_STREAM_STATE_RESERVED_REMOTE: 4,
+  NGHTTP2_STREAM_STATE_HALF_CLOSED_LOCAL: 5,
+  NGHTTP2_STREAM_STATE_HALF_CLOSED_REMOTE: 6,
+  NGHTTP2_STREAM_STATE_CLOSED: 7,
+  NGHTTP2_HCAT_REQUEST: 0,
+  NGHTTP2_HCAT_RESPONSE: 1,
+  NGHTTP2_HCAT_PUSH_RESPONSE: 2,
+  NGHTTP2_HCAT_HEADERS: 3,
+  NGHTTP2_NO_ERROR: 0,
+  NGHTTP2_PROTOCOL_ERROR: 1,
+  NGHTTP2_INTERNAL_ERROR: 2,
+  NGHTTP2_FLOW_CONTROL_ERROR: 3,
+  NGHTTP2_SETTINGS_TIMEOUT: 4,
+  NGHTTP2_STREAM_CLOSED: 8,
```

Comment:
I noticed that `require('http2').constants` grabs values from both [`nghttp2_error_code`](https://github.com/nodejs/http2/blob/0cb5d518dddcc8a25101ece374b43c158833f9e6/deps/nghttp2/lib/includes/nghttp2/nghttp2.h) and [`nghttp2_stream_flags`](https://github.com/nodejs/http2/blob/7ab1eba688e6e958176a2490ba033676081f15fb/src/node_http2_core.h#L57), which contain clashing values for `NGHTTP2_STREAM_CLOSED`. Should one of these enums have their values changed?